### PR TITLE
Feature/campaign

### PR DESCRIPTION
**Pull request by Stanislav123**
I want to merge this change because...

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by marianmazarovici** [comment]
I'm out of the office until next week. Gabriele is off too. He can review
your PR when he will return on Thursday. Meanwhile, continue to append new
commits to the PR.

*Best regards,*

<http://vonq.com/>
*Marian Mazarovici*
*Developer*

+44 7450 203 968
marian@VONQ.com
www.VONQ.com <http://www.vonq.com/>
<https://www.linkedin.com/company/vonq/>  <https://twitter.com/vonq>
<https://www.instagram.com/vonq_hq/>  <https://www.facebook.com/VONQ.NL/>
<https://www.xing.com/companies/vonq>
*SMART RECRUITMENT MARKETING*

**Comment by marianmazarovici** [comment]
Meanwhile, have a look at the Github build - it's failing :)

File: saleor/graphql/campaign/enums.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+from enum import Enum
+
+
+class IndustryEnum(Enum):
```
Comment (marianmazarovici) [inline saleor/graphql/campaign/enums.py]: Industries need to be values from this endpoint https://pkb.web-production.vonq-aws.com/products/industries/
The labels are in the `name` key. The raw values passed to the order are in the `id` key

File: saleor/graphql/campaign/enums.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    RETAIL = "RETAIL"
+    SCIENCE_RESEARCH = "SCIENCE_RESEARCH"
+    SUSTAINABILITY = "SUSTAINABILITY"
+    TELECOMMUNICATIONS = "TELECOMMUNICATIONS"
+    TRANSPORT_AND_RAIL = "TRANSPORT_AND_RAIL"
+    TRAVEL_AND_TOURISM = "TRAVEL_AND_TOURISM"
+
+
+class SeniorityEnum(Enum):
+    TRAINEE = "TRAINEE"
```
Comment (marianmazarovici) [inline saleor/graphql/campaign/enums.py]: seniorities need to be from below. labels are from the `name.value` where languageCode is `en_EN`. raw value passed to the order needs to be from the `id` key

```
[
    {
        "id": 1,
        "name": [
            {
                "languageCode": "de_DE",
                "value": "Geschäftsführer / C-Level"
            },
            {
                "languageCode": "en_GB",
                "value": "Executive/Director"
            },
            {
                "languageCode": "nl_NL",
                "value": "Executive/Director"
            }
        ]
    },
    {
        "id": 2,
        "name": [
            {
                "languageCode": "de_DE",
                "value": "Manager"
            },
            {
                "languageCode": "en_GB",
                "value": "Manager"
            },
            {
                "languageCode": "nl_NL",
                "value": "Manager"
            }
        ]
    },
    {
        "id": 3,
        "name": [
            {
                "languageCode": "de_DE",
                "value": "Professional"
            },
            {
                "languageCode": "en_GB",
                "value": "Mid-Senior level"
            },
            {
                "languageCode": "nl_NL",
                "value": "Professional"
            }
        ]
    },
    {
        "id": 4,
        "name": [
            {
                "languageCode": "de_DE",
                "value": "Berufseinsteiger"
            },
            {
                "languageCode": "en_GB",
                "value": "Entry level/Graduate"
            },
            {
                "languageCode": "nl_NL",
                "value": "Entry level/Graduate"
            }
        ]
    },
    {
        "id": 5,
        "name": [
            {
                "languageCode": "de_DE",
                "value": "Student/Trainee"
            },
            {
                "languageCode": "en_GB",
                "value": "Student/Trainee"
            },
            {
                "languageCode": "nl_NL",
                "value": "Student/Trainee"
            }
        ]
    }
]
```

File: saleor/graphql/campaign/enums.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    GRADUATE = "GRADUATE"
+    MID_OR_SENIOR = "MID_OR_SENIOR"
+    MANAGER = "MANAGER"
+
+
+class EducationLeavelEnum(Enum):
```
Comment (marianmazarovici) [inline saleor/graphql/campaign/enums.py]: same as for seniority

```
[
    {
        "id": 1,
        "name": [
            {
                "languageCode": "de_DE",
                "value": "Master / PhD / Doktor"
            },
            {
                "languageCode": "en_GB",
                "value": "Master / Post-Graduate / PhD"
            },
            {
                "languageCode": "nl_NL",
                "value": "Master / Postdoctoraal"
            }
        ]
    },
    {
        "id": 2,
        "name": [
            {
                "languageCode": "de_DE",
                "value": "Bachelor / Absolvent"
            },
            {
                "languageCode": "en_GB",
                "value": "Bachelor / Graduate"
            },
            {
                "languageCode": "nl_NL",
                "value": "HBO / Bachelor"
            }
        ]
    },
    {
        "id": 3,
        "name": [
            {
                "languageCode": "de_DE",
                "value": "Ausbildung"
            },
            {
                "languageCode": "en_GB",
                "value": "Vocational / Diploma / Associates degree"
            },
            {
                "languageCode": "nl_NL",
                "value": "MBO / Beroepssecundair onderwijs"
            }
        ]
    },
    {
        "id": 4,
        "name": [
            {
                "languageCode": "de_DE",
                "value": "Haupt-/ Realschulabschluss"
            },
            {
                "languageCode": "en_GB",
                "value": "GCSE / A-Level / Highschool / GED"
            },
            {
                "languageCode": "nl_NL",
                "value": "Middelbaar / Secundair onderwijs"
            }
        ]
    }
]
```

**Review by marianmazarovici — CHANGES_REQUESTED** [review]

File: saleor/campaign/models.py [hunk 4]
```diff
@@ -14,3 +16,23 @@ class Campaign(models.Model):
...
     education = models.CharField(max_length=40)
     job_function = models.CharField(max_length=255)
+    created = models.DateTimeField(auto_now_add=True)
+
+
+class JobInfo(models.Model):
+    created = models.DateTimeField(auto_now_add=True)
+    title = models.CharField(max_length=250, blank=True)
+    campaign = models.OneToOneField('campaign.Campaign', related_name='job_info', on_delete=models.CASCADE)
+    industry = models.CharField(max_length=50)
```
Comment (marianmazarovici) [inline saleor/campaign/models.py]: Makes sense for industry, education, seniority and job function to be part of the JobInfo model. Why are they also present in the Campaign model?

File: saleor/graphql/campaign/enums.py [hunk 5]
```diff
@@ -62,3 +62,133 @@ class EducationLeavelEnum(Enum):
...
+    FIXED_TERM = "FIXED_TERM"
+    FIXED_TERM_WITH_PERMAMENT = "FIXED_TERM_WITH_PERMAMENT"
+    FREELANCE = "FREELANCE"
+    INTERSHIP = "INTERSHIP"
+    PERMAMENT = "PERMAMENT"
+    TRAINEESHIP = "TRAINEESHIP"
+
+
+class CurrenciesEnum(Enum):
+    AFN = "AFN"
```
Comment (marianmazarovici) [inline saleor/graphql/campaign/enums.py:83]: Can you confirm these are https://en.wikipedia.org/wiki/ISO_4217 codes?

Comment (gabalese) [reply saleor/graphql/campaign/enums.py:83]: Quite sure there's several python libraries that bundle this enum.

**Comment by gabalese** [comment]
I see we're effectively adding a new campaign app – which makes sense, but does that mean that there's some effort involved in syncing the changes with upcoming new releases from Saleor?

**Pull request closed without merging**